internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -98,21 +98,16 @@ func (r *Reconciler) reconcileClusterShim(ctx context.Context, s *scope.Scope) e
 	// creating InfrastructureCluster/ControlPlane objects and updating the Cluster with the
 	// references to above objects.
 	if s.Current.InfrastructureCluster == nil || s.Current.ControlPlane.Object == nil {
-		// Note: we are using Patch instead of create for ensuring consistency in managedFields for the entire controller
-		// but in this case it isn't strictly necessary given that we are not using server side apply for modifying the
-		// object afterwards.
-		helper, err := r.patchHelperFactory(ctx, nil, shim)
-		if err != nil {
-			return errors.Wrap(err, "failed to create the patch helper for the cluster shim object")
-		}
-		if err := helper.Patch(ctx); err != nil {
-			return errors.Wrap(err, "failed to create the cluster shim object")
-		}
-
-		if err := r.APIReader.Get(ctx, client.ObjectKeyFromObject(shim), shim); err != nil {
-			return errors.Wrap(err, "get shim after creation")
+		// Given that the cluster shim is a temporary object which is only modified
+		// by this controller, it is not necessary to use the SSA patch helper.
```

Comment:
Did a bit of testing with Fabrizio and we think that it would be just simpler to use the regular client here (as documented in the godoc comment)